File: src/main/java/play/pluv/playlist/domain/MusicStreaming.java
package play.pluv.playlist.domain;

import static play.pluv.playlist.exception.PlayListExceptionType.PLAYLIST_PROVIDER_NOT_FOUND;

import java.util.Arrays;
import lombok.Getter;
import play.pluv.playlist.exception.PlayListException;

@Getter
public enum MusicStreaming {

  SPOTIFY("spotify"),
  YOUTUBE("youtube"),
  APPLE("apple");

  private final String name;

  MusicStreaming(final String name) {
    this.name = name;
  }

  public static MusicStreaming from(final String name) {
    return Arrays.stream(values())
        .filter(streaming -> streaming.getName().equals(name))
        .findAny()
        .orElseThrow(() -> new PlayListException(PLAYLIST_PROVIDER_NOT_FOUND));
  }
}

File: src/main/kotlin/play/pluv/playlist/domain/MusicStreaming.kt
package play.pluv.playlist.domain

import play.pluv.playlist.exception.PlayListException
import play.pluv.playlist.exception.PlayListExceptionType

enum class MusicStreaming(name: String) {
    SPOTIFY("spotify"),
    YOUTUBE("youtube"),
    APPLE("apple");

    companion object {
        fun from(name: String): MusicStreaming {
            return entries.find { it.name == name }
                ?: throw PlayListException(PlayListExceptionType.PLAYLIST_PROVIDER_NOT_FOUND)
        }
    }
}